ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/MachineAccessIntegrationForm.tsx

```diff
@@ -0,0 +1,377 @@
+import FormMessage from 'Components/PatternFly/FormMessage';
+import FormLabelGroup from 'Containers/Integrations/IntegrationForm/FormLabelGroup';
+import IntegrationFormActions from 'Containers/Integrations/IntegrationForm/IntegrationFormActions';
+import FormSaveButton from 'Components/PatternFly/FormSaveButton';
+import FormCancelButton from 'Components/PatternFly/FormCancelButton';
+import SelectSingle from 'Components/SelectSingle';
+import { fetchRolesAsArray, Role } from 'services/RolesService';
+import { MachineConfigType } from 'services/MachineAccessService';
+
+export type MachineAccessConfig = {
+    id: string;
+    type: MachineConfigType;
+    tokenExpirationDuration: string;
+    mappings: {
+        key: string;
+        valueExpression: string;
+        role: string;
+    }[];
+    issuer: string;
+};
+
+export const validationSchema = yup.object().shape({
+    type: yup.string().trim().required('Type is required.'),
+    tokenExpirationDuration: yup.string().trim().required('Token expiration duration is required'),
+    mappings: yup.array().of(
+        yup.object().shape({
+            key: yup.string().trim().required('Key is required.'),
+            valueExpression: yup.string().trim().required('Value expression is required.'),
+            role: yup.string().trim().required('Role is required.'),
+        })
+    ),
+    issuer: yup.string().trim().required('Issuer is required.'),
+});
+
+export const defaultValues: MachineAccessConfig = {
+    issuer: '',
+    mappings: [],
+    tokenExpirationDuration: '',
+    type: 'GENERIC',
+    id: '',
+};
+
+function MachineAccessIntegrationForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<MachineAccessConfig>): ReactElement {
+    const formInitialValues = { ...defaultValues, ...initialValues };
+    const formik = useIntegrationForm<MachineAccessConfig>({
+        initialValues: formInitialValues,
+        validationSchema,
+    });
+    const {
+        values,
+        touched,
+        errors,
+        dirty,
+        isValid,
+        setFieldValue,
+        handleBlur,
+        isSubmitting,
+        isTesting,
+        onSave,
+        onCancel,
+        message,
+    } = formik;
+
+    function onChange(value, event) {
+        return setFieldValue(event.target.id, value);
+    }
+
+    const [roles, setRoles] = useState<Role[]>([]);
+    const [alertRoles, setAlertRoles] = useState<ReactElement | null>(null);
+
+    useEffect(() => {
+        fetchRolesAsArray()
+            .then((rolesFetched) => {
+                setRoles(rolesFetched);
+                setAlertRoles(null);
+            })
+            .catch((error) => {
+                setAlertRoles(
+                    <Alert title="Fetch roles failed" variant={AlertVariant.warning} isInline>
+                        {error.message}
+                    </Alert>
+                );
+            });
```

Comment:
Thanks for handling the sad-path of this API call.

small nit: the UI team has adopted a "fail-safe" function to ensuring consistent error messages

You can see an example in this recent file, [AdministrationEventsPage](https://github.com/stackrox/stackrox/blob/0a7614a0d60478d568cdabd0f0b2e7e507db5124/ui/apps/platform/src/Containers/Administration/Events/AdministrationEventsPage.tsx#L58-L62).

Also, note that the UI team is also trying to standardize on setting just the error message itself in state, and then conditionally rendering the <Alert> component in the template if an error message is set.

Again, see Mark's recent example above, and [this conditionally-rendered part of the TSX](https://github.com/stackrox/stackrox/blob/0a7614a0d60478d568cdabd0f0b2e7e507db5124/ui/apps/platform/src/Containers/Administration/Events/AdministrationEventsPage.tsx#L103-L119) lower in that same file.

```suggestion
            .catch((error) => {
                setAlertRoles(
                    <Alert title="Fetch roles failed" variant={AlertVariant.warning} isInline>
                        {getAxiosErrorMessage(error)}
                    </Alert>
                );
            });
```